app/assets/javascripts/app/frontend/controllers/editor.js

```diff
@@ -15,16 +15,22 @@ angular.module('app.frontend')
 
       link:function(scope, elem, attrs, ctrl) {
 
+        /**
+         * Insert 4 spaces when a tab key is pressed, 
+         * only used when inside of the text editor.
+         */
+        var handleTab = function (event) {
+          if (event.which == 9) {
+            event.preventDefault();
+            var start = event.target.selectionStart;
+            var end = event.target.selectionEnd;
```

Comment:
Bah, I found another bug. 

If you try add a tab when you have more text after it. (i.e. in the middle of the note) it jumps to the end of the document. I think its because of this hack. Will look into this.